rewatch/src/project_context.rs

```diff
@@ -0,0 +1,288 @@
+                                parent_config: workspace_config,
+                                parent_path: workspace_rescript_json.clone(),
+                            })
+                        } else {
+                            // There is a parent rescript.json, but it has no reference to the current package.
+                            // However, the current package could still be a monorepo root!
+                            monorepo_or_single_project(&path, current_rescript_json, current_config)
+                        }
+                    }
+                }
+            }
+        };
+        context.iter().for_each(|pc| {
+            debug!("Created project context {:#?} for \"{}\"", pc, path.display());
+        });
+        context
+    }
+
+    pub fn get_root_config(&self) -> &Config {
+        match self {
+            ProjectContext::SingleProject { config, .. } => config,
+            ProjectContext::MonorepoRoot { config, .. } => config,
+            ProjectContext::MonorepoPackage {
+                parent_config: config,
+                ..
+            } => config,
+        }
+    }
+
+    pub fn get_root_path(&self) -> &Path {
+        let rescript_json = match self {
+            ProjectContext::SingleProject { path, .. } => path,
+            ProjectContext::MonorepoRoot { path, .. } => path,
+            ProjectContext::MonorepoPackage {
+                parent_path: path, ..
+            } => path,
+        };
+        rescript_json.parent().unwrap()
+    }
+
+    pub fn get_suffix(&self) -> String {
+        match self {
+            ProjectContext::SingleProject { config, .. }
+            | ProjectContext::MonorepoRoot { config, .. }
+            | ProjectContext::MonorepoPackage {
+                parent_config: config,
+                ..
+            } => config.suffix.clone().unwrap_or(String::from(".res.mjs")),
+        }
+    }
+
+    /// Returns the local packages relevant for the current context.
+    /// Either a single project, all projects from a monorepo or a single package inside a monorepo.
+    pub fn get_scoped_local_packages(&self, include_dev_deps: bool) -> AHashSet<String> {
+        let mut local_packages = AHashSet::<String>::new();
+        match &self {
+            ProjectContext::SingleProject { config, .. } => {
+                local_packages.insert(config.name.clone());
+            }
+            ProjectContext::MonorepoRoot {
+                config,
+                local_dependencies,
+                local_dev_dependencies,
+                ..
+            } => {
+                local_packages.insert(config.name.clone());
+                for dep in local_dependencies {
+                    local_packages.insert(dep.clone());
+                }
+                if include_dev_deps {
+                    for dep in local_dev_dependencies {
+                        local_packages.insert(dep.clone());
+                    }
+                }
```

Comment:
I think maybe there is an `extend` function or something like that.
I'm not sure what the underlying representation of the hashset is, but a bulk operation will be more optimized. So probably something like:

```
let deps = local_dev_dependencies.into_iter().collect::AHashSet<String>();
local_packages.extend(deps)
```

Is faster because the CPU can plan ahead better.